xbmc/XBDateTime.cpp

```diff
@@ -818,13 +818,15 @@ void CDateTime::GetAsTm(tm& time) const
   KODI::TIME::SystemTime st;
   GetAsSystemTime(st);
 
+  time = {};
   time.tm_year = st.year - 1900;
   time.tm_mon = st.month - 1;
   time.tm_wday = st.dayOfWeek;
   time.tm_mday = st.day;
   time.tm_hour = st.hour;
   time.tm_min = st.minute;
   time.tm_sec = st.second;
+  time.tm_isdst = -1;
```

Comment:
My only query about this PR is this change.

https://github.com/xbmc/xbmc/pull/21660

The blanket -1 was said to not really be accurate, and may be better off with some time zone bias applied.

Reply:
I'm 99 percent certain that this change is correct :smiley: 

First, lets see what's wrong with the original code. To observe the issue you can set you PCs time zone to something in the southern hemisphere (I tried Auckland). If you then run the unit tests you will see these tests failing:
```
[  FAILED  ] TestDateTime.TmOperators
[  FAILED  ] TestDateTime.GetAsTm
```
What goes wrong? Let's look at the `GetAsTm` test:
https://github.com/xbmc/xbmc/blob/9104aa0af563f06dd6e4ed763b90f570ab556941/xbmc/test/TestDateTime.cpp#L543-L552
Here `CDateTime` is set to a time in May which doesn't observe daylight saving time in the southern hemisphere. When now calling `CDateTime::getAsTm` due to the missing initialization the `tm::tm_isdst` is likely some value > 0. Then `mktime` is called to normalize the `struct tm`. Because `tm::tm_isdst > 0` but no daylight saving active at the specified date, `mktime` subtracts an hour and sets `tm_isdst` to 0. But the `KODI::TIME::SystemTime` already was accounting for non daylight saving and we are now one hour off.

By initializing `tm_isdst` to `-1` we tell ``mktime to not adjust the time and to only set `tm_isdst` to the correct value.

I'll extend the unit tests so that this is caught independently of the local timezone in which the tests are run.

Reply:
I'd postpone this pretty minor fix until we get cdatetime topic in. I have next week full with paid work but after that let me fix the freebsd failing test.

Reply:
@neo1973 can you please have a look at cdatetime topic (without rebase) and check if anything from this PR is applicable there?

Reply:
> I'd postpone this pretty minor fix until we get cdatetime topic in. I have next week full with paid work but after that let me fix the freebsd failing test.

While the CDateTime changes will make this commit obsolete I still think it doesn't hurt do merge it now. Also it would make for a cleaner backport if this first goes to master.

> can you please have a look at cdatetime topic (without rebase) and check if anything from this PR is applicable there?

The only overlap is the `CDateTime::GetAsTm(tm& time)` change and that will be replaced by your changes.

Reply:
OK, thanks for checks! Then let's make this PR in. I thought the overlap will be huge in TestDateTime :)